sparse/tests/test_core.py

```diff
@@ -359,15 +358,31 @@ def test_addition_not_ok_when_large_and_sparse():
         np.exp(x)
 
 
-@pytest.mark.xfail
-def test_addition_broadcasting():
-    x = random_x((2, 3, 4))
+@pytest.mark.parametrize('func', [operator.add, operator.mul])
+@pytest.mark.parametrize('shape1,shape2', [((2, 3, 4), (3, 4)),
+                                           ((3, 4), (2, 3, 4)),
+                                           ((3, 1, 4), (3, 2, 4)),
+                                           ((1, 3, 4), (3, 4)),
+                                           ((3, 4, 1), (3, 4, 2)),
+                                           ((1, 5), (5, 1))])
+def test_broadcasting(func, shape1, shape2):
+    x = random_x(shape1)
     a = COO.from_numpy(x)
 
-    z = random_x((3, 4))
+    z = random_x(shape2)
     c = COO.from_numpy(z)
 
-    assert_eq(x + z, a + c)
+    assert_eq(func(x, z), func(a, c))
+
+
+@pytest.mark.parametrize('shape1,shape2', [((3, 4), (2, 3, 4)),
+                                           ((3, 1, 4), (3, 2, 4)),
+                                           ((3, 4, 1), (3, 4, 2))])
+def test_broadcast_to(shape1, shape2):
+    x = random_x(shape1)
+    a = COO.from_numpy(x)
+
+    assert_eq(np.broadcast_to(x, shape2), a.broadcast_to(shape2))
```

Comment:
These tests are quite nice and establishing confidence!  Are there any other things we should check on the resulting arrays?  For example the number of non-zeroes, if sortedness is preserved (or should it be?) etc..

Reply:
Well, the number of non-zeros is already minimal (see lines 730, 731). I could add that to the test, if you like.

As a closing note, do I close this pull request and open a new one when I've done the changes, will the changes be automatically merged into this pull request, ...? I hope you don't mind me asking, this is actually my first time contributing to a project without being approved, and I've only ever made one pull request before.

Reply:
> Well, the number of non-zeros is already minimal (see lines 730, 731). I could add that to the test, if you like.

Yes, you should definitely test things like this.  You may know that it's true now, but other developers may come by and change your code.  You want them to see failed tests if they change something important.

Reply:
It was nice you suggested that we test the `nnz`. Apparently, it was failing for shapes `(5, 1)` and `(1, 5)`. Not sure why yet.

Reply:
Did this get resolved?  At first glance it seems fine:

```python
In [26]: x = sparse.COO(random_x((5, 1)))

In [27]: x
Out[27]: <COO: shape=(5, 1), dtype=float64, nnz=3, sorted=True, duplicates=False>

In [28]: x.broadcast_to((5, 5))
Out[28]: <COO: shape=(5, 5), dtype=float64, nnz=15, sorted=True, duplicates=False>
```

Adding it explicitly to tests to avoid future regression might be useful

Reply:
Yes, it was resolved. It wasn't `broadcast_to` that was failing, it was `elemwise_binary` where one element was `(1, 5)` and another was `(5, 1)`, or vice versa. I realized it was a wider bug only showing up with those specific shapes, and I made sure it won't occur. It was fixed in 63e68d93827b3fa3dd29975bab4949787617b056.

Reply:
I don't think regression will be an issue because I replaced the logic where the bug was occurring entirely.